src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,256 @@
+and substitution instead.
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [Jordan, von Neumann and Wigner, 1934][jordanvonneumannwigner1934]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+
+-/
+
+set_option old_structure_cmd true
+
+section non_unital_non_assoc_ring
+
+variables {A : Type*} [non_unital_non_assoc_semiring A]
+
+namespace non_unital_non_assoc_ring
+/--
+Left multiplication operator
+-/
+@[simps] def L : A →+ add_monoid.End A := add_monoid_hom.mul
+
+/--
+Right multiplication operator
+-/
+@[simps] def R : A→+(add_monoid.End A) := add_monoid_hom.flip (L : A →+ add_monoid.End A)
+
+lemma L_def (a b : A) : L a b = a*b := rfl
+
+lemma R_def (a b : A) : R a b = b*a := rfl
+
+end non_unital_non_assoc_ring
+
+end non_unital_non_assoc_ring
+
+open non_unital_non_assoc_ring
+/--
+A non unital, non-associative ring with a (non-commutative) Jordan multiplication.
+-/
+class jordan_ring(A : Type*) [non_unital_non_assoc_ring A] :=
+(lmul_comm_rmul : ∀ a b : A, (a * b) * a = a * (b * a))
+(lmul_lmul_comm_lmul: ∀ a b : A, (a * a) * (a * b) = a * ((a * a) * b))
+(lmul_lmul_comm_rmul: ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
+(lmul_comm_rmul_rmul: ∀ a b : A, (a * b) * (a * a) = a * (b * (a * a)))
+(rmul_comm_rmul_rmul: ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
+
+lemma jordan_operators (A : Type*) [non_unital_non_assoc_ring A] [jordan_ring A] (a : A) :
+(⁅L a, R a⁆ = 0) ∧ (⁅L a, L (a * a)⁆ = 0) ∧ (⁅L a, R (a * a)⁆ = 0) ∧ (⁅L (a * a), R a⁆ = 0) ∧
+(⁅R a, R (a * a)⁆ = 0) :=
+begin
+  rw [ring.lie_def, ring.lie_def, ring.lie_def, ring.lie_def, ring.lie_def ],
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_comm_rmul, rw sub_self, },
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_lmul_comm_lmul, rw sub_self, },
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_comm_rmul_rmul, rw sub_self, },
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_lmul_comm_rmul, rw sub_self, },
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.rmul_comm_rmul_rmul, rw sub_self, },
+end
```

Comment:
This would be much better as 5 separate lemmas

Reply:
Done.